libs/state/actions/src/lib/inject.ts

```diff
@@ -0,0 +1,32 @@
+import { DestroyRef, ErrorHandler, inject } from '@angular/core';
+import { RxActionFactory } from './actions.factory';
+import { ActionTransforms, RxActions } from './types';
+
+export function withTransforms<
+  Actions extends object,
+  Transforms extends ActionTransforms<Actions> = {}
+>(transforms: Transforms) {
+  return (rxActionFactory: RxActionFactory<Actions>) =>
+    rxActionFactory.create<Transforms>(transforms);
+}
+
+export function rxActions<
+  Actions extends object,
+  Transforms extends ActionTransforms<Actions> = {}
+>(
+  withTransformsFn?: (
+    rxActionFactory: RxActionFactory<Actions>
+  ) => RxActions<Actions, Transforms>
+): RxActions<Actions, Transforms> {
+  const errorHandler = inject(ErrorHandler, { optional: true }) ?? undefined;
+  const rxActionFactory = new RxActionFactory<Actions>(errorHandler);
```

Comment:
@BioPhoton I think we should refactor the RxActionFactory and use the inject function as well for the error handler, I'm not sure to get why it is optional tho.

Reply:
The reason that is it optional is if I don't pass anything, Angular's default ErrorHandler is used

Reply:
I refactored the ErrorHandler usage in RxActionFactory and RxEffects to use inject instead of DI through ctors in [07a916b](https://github.com/rx-angular/rx-angular/pull/1543/commits/07a916b36c1d27ce569d7ad3919ebcf5dd6ef48e). However, I kept the provider optional to not break the previous behavior.